Lib/asyncio/timeouts.py

```diff
@@ -0,0 +1,141 @@
+import enum
+
+from types import TracebackType
+from typing import final, Any, Dict, Optional, Type
+
+from . import events
+from . import tasks
+
+
+__all__ = (
+    "Timeout",
+    "timeout",
+    "timeout_at",
+)
+
+
+class _State(enum.Enum):
+    CREATED = "created"
+    ENTERED = "active"
+    EXPIRING = "expiring"
+    EXPIRED = "expired"
+    EXITED = "finished"
+
+
+@final
+class Timeout:
+
+    def __init__(self, when: Optional[float]) -> None:
+        self._state = _State.CREATED
+
+        self._timeout_handler: Optional[events.TimerHandle] = None
+        self._task: Optional[tasks.Task[Any]] = None
+        self._when = when
+
+    def when(self) -> Optional[float]:
+        return self._when
+
+    def reschedule(self, when: Optional[float]) -> None:
+        assert self._state is not _State.CREATED
+        if self._state is not _State.ENTERED:
+            raise RuntimeError(
+                f"Cannot change state of {self._state.value} Timeout",
+            )
+
+        self._when = when
+
+        if self._timeout_handler is not None:
+            self._timeout_handler.cancel()
+
+        if when is None:
+            self._timeout_handler = None
+        else:
+            loop = events.get_running_loop()
+            self._timeout_handler = loop.call_at(
+                when,
+                self._on_timeout,
+            )
+
+    def expired(self) -> bool:
+        """Is timeout expired during execution?"""
+        return self._state in (_State.EXPIRING, _State.EXPIRED)
+
+    def __repr__(self) -> str:
+        info = ['']
+        if self._state is _State.ENTERED:
+            info.append(f"when={self._when:.3f}")
```

Comment:
This will crash if `when` is None. Test:
```py
    async with asyncio.timeout(None) as t:
        print(repr(t))
```

Reply:
Fixed. Now it displays `'when=None'` for `None` timeout.